registry/api/v2/urls.go

```diff
@@ -42,32 +42,46 @@ func NewURLBuilderFromString(root string) (*URLBuilder, error) {
 }
 
 // NewURLBuilderFromRequest uses information from an *http.Request to
-// construct the root url.
-func NewURLBuilderFromRequest(r *http.Request) *URLBuilder {
-	var scheme string
-
-	forwardedProto := r.Header.Get("X-Forwarded-Proto")
-
-	switch {
-	case len(forwardedProto) > 0:
-		scheme = forwardedProto
-	case r.TLS != nil:
-		scheme = "https"
-	case len(r.URL.Scheme) > 0:
-		scheme = r.URL.Scheme
-	default:
-		scheme = "http"
-	}
+// construct the root url. configScheme and configHost are the URI scheme and
+// hostname, respectively, from the configuration - or empty strings if
+// unspecified.
+func NewURLBuilderFromRequest(r *http.Request, configScheme, configHost string) *URLBuilder {
```

Comment:
I'm confused about why extracting the path from the request isn't necessary. `URLBuilder` does use the path when building a URL.

Maybe you're suggesting that the user should provide a complete URL prefix in the config, instead of having the path portion extracted from the request? Currently only the URL scheme and hostname are used (thus the name `httpHost`). I'd be fine with changing this parameter to specify the path as well, if you think that's the correct approach. As a user, I think I'd find it easier to just enter `https://myregistry.net` instead of `https://myregistry.net/v2/`, but if we document it sufficiently, I think it would be fine.